code/procs/helpers.dm

```diff
@@ -1879,6 +1879,9 @@ proc/countJob(rank)
 	if (!allow_dead_antags && (!G.mind || G.mind && (G.mind.dnr || !isnull(G.mind.special_role) || G.mind.former_antagonist_roles.len))) // Dead antagonists have had their chance.
```

Comment:
no reason to check for DNR twice if you're adding in the new check. Half of this if can be scrapped, it should really just check the flag for allowing antags and if the mob has any of the antag indicators

and while we're at it kill of the .len check in favor of a length
```dm
length(G.mind.former_antagonist_roles)
```